src/BenchmarksServer/Startup.cs

```diff
@@ -1391,54 +1424,106 @@ private static Process StartProcess(string hostname, string benchmarksRepo, Serv
                     Log.WriteLine(e.Data);
                     standardOutput.AppendLine(e.Data);
 
-                    if (job.State == ServerState.Starting && (e.Data.ToLowerInvariant().Contains("started") || e.Data.ToLowerInvariant().Contains("listening")))
+                    if (job.State == ServerState.Starting &&
+                        (e.Data.ToLowerInvariant().Contains("started") || e.Data.ToLowerInvariant().Contains("listening")))
                     {
-                        job.StartupMainMethod = stopwatch.Elapsed;
+                        MarkAsRunning(hostname, benchmarksRepo, job, perfview, stopwatch, process);
+                    }
+                }
+            };
 
-                        Log.WriteLine($"Running job '{job.Id}' with scenario '{job.Scenario}'");
-                        job.Url = ComputeServerUrl(hostname, job);
+            stopwatch.Start();
+            process.Start();
+            process.BeginOutputReadLine();
 
-                        // Start perfview?
-                        if (perfview)
-                        {
-                            job.PerfViewTraceFile = Path.Combine(job.BasePath, "benchmarks.etl");
-                            var perfViewArguments = new Dictionary<string, string>();
-                            perfViewArguments["AcceptEula"] = "";
-                            perfViewArguments["NoGui"] = "";
-                            perfViewArguments["Process"] = process.Id.ToString();
+            if (iis)
+            {
+                await WaitToListen(job, hostname);
```

Comment:
This seems even better than waiting for stdout from Kestrel. Maybe we should use this logic for all servers.